### PR TITLE
Prerequisites for git submodule fixes

### PR DESCRIPTION
**Pull request by ausfeldt**

### CODE REVIEW

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Microsoft.Azure.Cosmos.Tests.csproj [hunk 1]
```diff
@@ -33,7 +33,7 @@
     <PackageReference Include="Moq" Version="4.8.3" />
     <PackageReference Include="MSTest.TestAdapter" Version="1.2.0" />
     <PackageReference Include="MSTest.TestFramework" Version="1.2.0" />
-    <PackageReference Include="Newtonsoft.Json" Version="10.0.2" />
+    <PackageReference Include="Newtonsoft.Json" Version="10.0.3" />
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Microsoft.Azure.Cosmos.Tests.csproj]: Looks odd that tests are using higher version. 
Can only master be updated use specific version?

File: Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosClientSideRequestStatistics.cs [hunk 2]
```diff
@@ -43,6 +36,12 @@ public CosmosClientSideRequestStatistics()
             this.RegionsContacted = new HashSet<Uri>();
         }
 
+        public List<Uri> ContactedReplicas { get; set; }
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosClientSideRequestStatistics.cs]: Won't this need new direct package?

File: Microsoft.Azure.Cosmos/src/Microsoft.Azure.Cosmos.csproj [hunk 3]
```diff
@@ -73,7 +73,7 @@
   </ItemGroup>
   
   <ItemGroup>
-    <PackageReference Include="Newtonsoft.Json" Version="10.0.2" />
+    <PackageReference Include="Newtonsoft.Json" Version="10.0.3" />
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Microsoft.Azure.Cosmos.csproj]: https://www.nuget.org/packages/Microsoft.Azure.Cosmos/ dependencies shows Newtonsoft.Json (>= 10.0.2).
It should remain at 10.0.2

File: Directory.Build.props [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (kirankumarkolli) [inline Directory.Build.props]: Not required. Its a search hint only not exact place to look-up.
Please check my private branch which works with-out it.

File: Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosClientSideRequestStatistics.cs [hunk 5]
```diff
@@ -24,13 +24,6 @@ internal sealed class CosmosClientSideRequestStatistics : IClientSideRequestStat
         internal List<StoreResponseStatistics> supplementalResponseStatisticsList;
         internal Dictionary<string, AddressResolutionStatistics> addressResolutionStatistics;
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosClientSideRequestStatistics.cs]: This change need to wait till direct refresh.

File: Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosClientSideRequestStatistics.cs (hunk 2 above)
Comment (ausfeldt) [reply Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosClientSideRequestStatistics.cs]: Yes, see line 244-245.

File: Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosClientSideRequestStatistics.cs (hunk 5 above)
Comment (ausfeldt) [reply Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosClientSideRequestStatistics.cs]: Correct

File: Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosClientSideRequestStatistics.cs [hunk 6]
```diff
@@ -24,13 +24,6 @@ internal sealed class CosmosClientSideRequestStatistics : IClientSideRequestStat
...
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosClientSideRequestStatistics.cs]: Can we please refresh the direct also with the required hotfix?

**Comment by kirankumarkolli** [comment]
Is conditional NewtonSoft dependent also needed?

**Comment by ausfeldt** [comment]
> Is conditional NewtonSoft dependent also needed?

No, assuming <PackageReference Update="Newtonsoft.Json" Version="10.0.3" /> in the mono repo does updates it like it should.

**Comment by kirankumarkolli** [comment]
@j82w, @simplynaveen20 please take a look

**Pull request merged**